resteasy-jaxrs/src/main/java/org/jboss/resteasy/core/AbstractAsynchronousResponse.java

```diff
@@ -38,6 +38,7 @@
    protected TimeoutHandler timeoutHandler;
    protected List<CompletionCallback> completionCallbacks = new ArrayList<CompletionCallback>();
    protected Map<Class<?>, Object> contextDataMap;
+   private boolean callbacksCalled;

```

Comment:
Should this be volatile?

Reply:
I don't think so, but honestly I'm not sure. At least I don't hope concurrent threads will resume the same request. Is that even specified?